azure-function-http/src/main/java/io/micronaut/azure/function/http/AzureHttpFunction.java

```diff
@@ -71,7 +72,7 @@ protected AzureHttpFunction() {
      */
     protected AzureHttpFunction(ApplicationContextBuilder applicationContextBuilder) {
         super(applicationContextBuilder);
-        httpHandler = new HttpHandler(getApplicationContext());
+        httpHandler = new HttpHandler(applicationContext, applicationContext.getBean(ConversionService.class));
```

Comment:
would not be better to keep `HttpHandler` constructor with a single parameter `ApplicationContext` and then do ?

```java
public HttpHandler(ApplicationContext applicationContext) {
    super(init(applicationContext), applicationContext.getBean(ConversionService.class));
}
```

Reply:
Good call, done 👍